### PR TITLE
Add ARM64 build

### PR DESCRIPTION
**Pull request by ciaranmcnulty**
I can't use the docker image on my M1 Mac under emulation due to a segfault[^1], so this is a start at multi-architecture build including ARM64 (see #11)

This does not (yet) lead to a successful build of the cucumber project, just a successful `docker build`. I'll work through the rest of the issues and see whether they need to be fixed in the other repo or this one, hence the draft status

 - [x] adds .dockerignore
 - [x] switch to buildx for easier multi-arch 
 - [x] change download URLs that are architecture-specific
 - [x] Javascript builds in monorepo
   - [x] remove Chrome if on ARM (no ARM build is available) 
   - [x] see if Chromium can replace Chrome (only used for html-formatter right now)
 - [x] fix the CI build (Matt broke it, not sure how)
 - [x] Java builds in monorepo (something about `JAVA_HOME` not set?)
 - [x] dotnet builds in monorepo
 - [x] Ruby builds in monorepo
 - [ ] cucumber-js builds
 - [ ] cucumber-jvm builds
 - [ ] cucumber-ruby builds

[^1]: It seems a long route to get this sort of thing fixed in QEMU

### CODE REVIEW

**Comment by ciaranmcnulty** [comment]
@mattwynne not sure why the build is failing on Intel - is probably downloading the wrong file(s) for dotnet

File: Makefile [hunk 1]
```diff
@@ -1,17 +1,12 @@
-NAME    := cucumber/cucumber-build
-VERSION := 0.2.0
+NAME      := cucumber/cucumber-build
+VERSION   := 0.2.0
```
Comment (ciaranmcnulty) [inline Makefile]: Why is this needed? Can we just omit the `—platforms` flag if the env var isn’t set?

Comment (mattwynne) [reply Makefile]: I think we need at least the version number checked in somewhere, for publishing the image.

File: Makefile [hunk 2]
```diff
@@ -1,28 +1,18 @@
-NAME    := cucumber/cucumber-build
-TAG     := 0.1.0
-IMG     := ${NAME}:${TAG}
-LATEST  := ${NAME}:latest
+NAME      := cucumber/cucumber-build
+VERSION   := 0.2.0
+DEFAULT_PLATFORM = $(shell [ $$(arch) = "arm64" ] && echo "linux/arm64" || echo "linux/amd64")
```
Comment (ciaranmcnulty) [inline Makefile:3]: There's actually a build arg for this already that includes the `linux/`

Comment (ciaranmcnulty) [reply Makefile:3]: https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope

File: Makefile [hunk 3]
```diff
@@ -42,3 +42,7 @@ docker-run-with-secrets: default
 	  -it ${NAME} \
 	  bash
 .PHONY: run-with-secrets
+<<<<<<< HEAD
```
Comment (ciaranmcnulty) [inline Makefile]: @mattwynne bad merge

File: Makefile (hunk 2 above)
Comment (mattwynne) [reply Makefile:3]: aye, but how could you use that here to give people a sensible default when they run `make`?

**Comment by mattwynne** [comment]
I've raised https://github.com/NixOS/nixpkgs/issues/122504 about the errors installing chromium. I'm wondering if we'll have to do without it for now, maybe just ignore these tests in the monorepo somehow.

**Comment by mattwynne** [comment]
Today I've tried to script the chrome download so we download the right binary for arm linux (see dfa5cf6)

When I try to run that binary, I see this:

```
cukebot@416564d37d54:/app$ ls -la /usr/bin/chromium-browser 
lrwxrwxrwx 1 root root 33 May 11 16:55 /usr/bin/chromium-browser -> /home/cukebot/chrome-linux/chrome
cukebot@416564d37d54:/app$ /usr/bin/chromium-browser 
/lib/ld-linux-armhf.so.3: No such file or directory
```

So I guess that either:

- I've picked the wrong binary
- We need some other stuff installed for the binary to work
- We need to do something less naive than just simlinking to the main binary program, and add some stuff to `$PATH`
- Something else

Any thoughts @ciaranmcnulty?

**Comment by mattwynne** [comment]
OK I *think* this is good to go.

I've tested it in the monorepo and there are still some issues building it on the arm platform, but I think that's now down to assumptions about the architecture in those build scripts, rather than anything wrong with this container.

It would be useful to test the `amd64` platform still works OK though I guess. Maybe I'll do that with a manual build/push and a PR on the monorepo.

**Comment by mattwynne** [comment]
I've manually pushed the container images to [cucumber/cucumber-build:pr-12](https://hub.docker.com/layers/cucumber/cucumber-build/pr-12/images/sha256-60be01ea805535d635211b671aa0bb75415d5792d2151990f18b9f8a4183d14a?context=explore)

**Comment by mattwynne** [comment]
OK this is released in v0.4.4 of the container. Huge thanks to @ciaranmcnulty for doing all the hard work to make this happen!

**Comment by ciaranmcnulty** [comment]
@mattwynne i think the commits tell a different story ;)

**Comment by mattwynne** [comment]
> @mattwynne i think the commits tell a different story ;)

It's quality not quantity that counts! I've learned loads about docker from working with you on this.
